### PR TITLE
compute before/after changes before applied extra edits, #86984

### PR DESCRIPTION
**Pull request by jrieken**
This PR cherry-picks the fix for https://github.com/microsoft/vscode/issues/86984 for the 1.41-recovery release.

### CODE REVIEW

**Comment by jrieken** [comment]
This is to call `this.getOverwriteInfo` *before* applying extra edits because the latter can change the editor position, something the former depends on.

File: src/vs/editor/contrib/suggest/suggestController.ts [hunk 1]
```diff
@@ -259,20 +259,21 @@ export class SuggestController implements IEditorContribution {
...
 		}
 
-		if (Array.isArray(suggestion.additionalTextEdits)) {
-			this.editor.executeEdits('suggestController.additionalTextEdits', suggestion.additionalTextEdits.map(edit => EditOperation.replace(Range.lift(edit.range), edit.text)));
-		}
+		// compute overwrite[Before|After] deltas BEFORE applying extra edits
+		const info = this.getOverwriteInfo(item, Boolean(flags & InsertFlags.AlternativeOverwriteConfig));
 
 		// keep item in memory
 		this._memoryService.memorize(model, this.editor.getPosition(), item);
```
Comment (alexr00) [inline src/vs/editor/contrib/suggest/suggestController.ts]: Why does this line also need to move? It looks like even in 1.40 it was after the `executeEdits`.

**Review by alexr00 — APPROVED** [review]

File: src/vs/editor/contrib/suggest/suggestController.ts (hunk 1 above)
Comment (jrieken) [reply src/vs/editor/contrib/suggest/suggestController.ts]: Yeah - that's a non minimal edit and fixes another (un-filed) issue where the memory service would also use the wrong editor position.

Comment (alexr00) [reply src/vs/editor/contrib/suggest/suggestController.ts]: LGTM

**Pull request merged**